openfisca_core/entities.py

```diff
@@ -51,9 +51,21 @@ def check_role_validity(self, role):
         if role is not None and not type(role) == Role:
             raise Exception("{} is not a valid role".format(role))
 
-    def __call__(self, variable_name, period, options = [], **parameters):
+    def __call__(self, variable_name, period=None, options = [], **parameters):
         self.check_variable_defined_for_entity(variable_name)
 
+        if period is None:
+            import traceback
+            stack = traceback.extract_stack()
+            filename, line_number, function_name, line_of_code = stack[-2]
+            raise ValueError(u'''
+You requested computation of variable "{}", but you did not specify on which period in "{}:{}":
+    {}
+You must always specify the period as the second parameter. The convention is to call this parameter "period". For example:
+    computed_salary = person('salary', period).
+See more information at <https://doc.openfisca.fr/coding-the-legislation/10_basic_example.html#basic-example>.
```

Comment:
As a legislation writer, this page doesn't tell me much about why I need to add a period as an argument. 
I'd reference [this page](https://doc.openfisca.fr/coding-the-legislation/35_periods.html), and maybe add a sentence there saying that within a formula, you must *always* specify a period when you compute another variable.